scipy/stats/_hypotests.py

```diff
@@ -379,18 +378,46 @@ def cramervonmises(rvs, cdf, args=()):
 
 def _get_wilcoxon_distr(n):
     """
-    Distribution of counts of the Wilcoxon ranksum statistic r_plus (sum of
-    ranks of positive differences).
-    Returns an array with the counts/frequencies of all the possible ranks
+    Distribution of probability of the Wilcoxon ranksum statistic r_plus (sum
+    of ranks of positive differences).
+    Returns an array with the probabilities of all the possible ranks
     r = 0, ..., n*(n+1)/2
     """
-    cnt = _wilcoxon_data.COUNTS.get(n)
+    c = np.ones(1, dtype=np.double)
+    for k in range(1, n + 1):
+        prev_c = c
+        c = np.zeros(k * (k + 1) // 2 + 1, dtype=np.double)
+        m = len(prev_c)
+        c[:m] = prev_c * 0.5
+        c[-m:] += prev_c * 0.5
+    return c
```

Comment:
```suggestion
    return c.astype(int)
```

Reply:
@mdhaber  c is a probabilities vector, I don't think it should be cast as int

Reply:
: ) sorry about that. I forgot we had switched to probabilities. Next time I won't review when I'm not thinking clearly : ) i was talking with @WarrenWeckesser about this PR and he'll take a look again soon, but it looks good to me.